x-pack/platform/plugins/shared/dataset_quality/server/routes/data_streams/get_data_streams_creation_date.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+import { dataStreamService } from '../../services';
+
+export async function getDataStreamsCreationDate({
+  esClient,
+  dataStreams,
+}: {
+  esClient: ElasticsearchClient;
+  dataStreams: string[];
+}) {
+  const matchingStreamsCreationDate = await Promise.all(
+    dataStreams.map(async (name) => {
+      const [dataStream] = await dataStreamService.getMatchingDataStreams(esClient, name);
+      const oldestIndex = dataStream?.indices[0];
+      if (!oldestIndex) {
+        return { name, creationDate: undefined };
+      }
+
+      const response = await esClient.cat.indices({
```

Comment:
While _cat api is not recommended for application use this is the only way to retrieve the creation date in serverless for now. We should change this once a proper approach exists (see https://github.com/elastic/elasticsearch-serverless/issues/3010)

Reply:
Could you please add a comment to the function to link this issue? If not we will forget to track the issue you mentioned

Reply:
added in https://github.com/elastic/kibana/pull/210301/commits/e80e806aaad837aa738dbf6f78ac038d50e0fb13